Assignment 5.Rmd

```diff
@@ -1,80 +1,122 @@
+### Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- rpart(formula = advice ~ prior_prob_count + prior_percent_correct + hints, 
+                     data = D1, 
+                     method = "class")
 ```
 
-#Plot tree
+### Plot tree
 ```{r}
-
+rpart.plot(x = score_ctree, yesno = 2, type = 0, extra = 0)
 ```
 
-Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
+Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?  
 
-#Test Tree
-Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
+The tree predicts what kinds of advice the teacher will give based on 3 attributes of the students: how many problems students have answered before, the percentage of those problems they got correct and how many hints they required. The tree have a depth of 5. The root node is `hints >= 2`.  
 
-```{r}
-#Upload new data
+The two behaviors teacher should most closely pay attemtion to are the number of hints the student requested and the percentage of those problems they got correct. If a student take too many hints or get too many incorrect answers, the teacher should intervene with the learning process of the student.  
 
-D2 <- 
+### Test Tree  
+Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students.   
 
-#Generate predicted advice using the predict() command for new students based on tree generated from old students
+```{r}
+# Upload new data
+D2 <- read.csv("intelligent_tutor_new.csv")
 
-D2$prediction <- 
+# Generate predictions on a test set
+D2$prediction <- predict(object = score_ctree, newdata = D2, type = "class")
 
+# Generate the actual values
+D2$actual <- ifelse(D2$score>=.8, "no action", 
+                    ifelse(D2$score<.8 & D2$score>=.6, "monitor", "intervene"))
 ``` 
+
 ## Part III
-Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?  
+
+All observed results only have one level - "no action" since all scores are 1, while predicted results have three levels - "no action", "monitor", "intervene".
+
+```{r}
+# Table
+table(D2$prediction, D2$actual)
```

Comment:
You are on the right track! Try to calculate the accuracy rate and describe the model in terms of its generalizability and accuracy rate.